### DISCUSSION TITLE
resume/retrain yolov11 with only 3 classes

### DISCUSSION THREAD

**Discussion by ardeal** [ultralytics/ultralytics#18420] [Q&A]
Hi,

I would like to resume training yolov11 model(yolo11m.pt) for only car, bicycle and person. 
How should I do?

The purpose of resuming training is that I would like to train yolov11 with the initialization of yolov11m.pt model and add my new images(care/bicycle/person).

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ardeal, thank you for your interest in Ultralytics 🚀! To help answer your question more effectively, I recommend providing some additional details such as the dataset structure, examples of your dataset labels (for car, bicycle, and person), and your training configuration. This will allow us to better understand your setup.

If this is a 🐛 Bug Report, please ensure you include a minimum reproducible example. This will significantly help in debugging the issue.

If this is a custom training ❓ Question, please provide logs or outputs from your training process and specify any adjustments you've made to the model or training hyperparameters. Also, verify that you're following best practices for training and data preparation to ensure quality results.

It’s a good idea to upgrade to the latest version of the `ultralytics` package, along with all its dependencies, to ensure compatibility and access to the latest features and fixes. You can do so using the appropriate package manager for your environment.

For development or training, confirmed working environments include notebooks with free GPU resources, preconfigured cloud environments such as GCP or AWS, and official Docker containers.

This is an automated response to guide you, but an Ultralytics engineer will review your question and assist you further soon 🚀!

**Comment by Y-T-G** [comment]
You can't do that. You can train on your dataset, but it's performance will get worse on other classes that are not in your dataset. And it will also get worse for the classes that are in your dataset if you only have few images due to overfitting.

**Comment by ardeal** [comment]
@Y-T-G 
Thank you for your reply!
I would like to re-explain my questions:
for default yolov11m.pt, there are 80 classes. 
for these 3 classes such as person, bicycle and car. I have already prepared more images for person, bicycle and car with correct label, and I would like to resume/retrain yolov11 network with yolov11m.pt initialization. The purpose of my training is to make yolov11 preform better.

**Reply by Y-T-G** [reply]
You can train normally. It's not resuming. It's starting from YOLO11 weights. But it will forget the old dataset it trained on if you don't include them in your dataset.

**Comment by ardeal** [comment]
@Y-T-G 
Thank you for your reply!
The problem I am experiencing is that the accuracy gets worse after I train yolov11 with my those 3 classes images. 
My training is initialized from the yolov11m.pt model. 

My question:
I add more images to continue the training. Normally, the accuracy should get better, why the accuracy get worse after my training?

**Reply by Y-T-G** [reply]
Because it overfitting to your dataset. The original dataset had 100k+ images. If you're training on a small dataset, the longer you train, the worse it will get.

**Reply by ardeal** [reply]
I should choose all those 3 classes images(person, bicycle, and car), and add my dataset to it. 
Then re-train yolov11 from the beginning.

**Reply by Y-T-G** [reply]
Do you want to keep other classes?

**Reply by ardeal** [reply]
I will do 3 things:
1) remove those images which don't contain person, car, bicycle.
2) change the label files of those person, car and bicycle images. (remove useless label, change useful label)
3) add my images(person, car, bicycle)
Then train yolov11 from the beginning.

Could you please tell me the training parameters(configurations) when you train yolov11m.pt?

**Reply by Y-T-G** [reply]
You don't need to train from the beginning. You just need more data. You should download the COCO dataset and then add your images.

**Reply by ardeal** [reply]
@Y-T-G 
Thank you!
I have download COCO, and trying to choose those images that I care about.
If I don't train from the beginning, where should I start from?
Could you please send me the training parameters(configurations)?

**Reply by Y-T-G** [reply]
You just use `model=yolo11m.pt`

You can also decrease the learning rate.

https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

**Reply by ardeal** [reply]
@Y-T-G 
Thank you for your reply!
I cannot access the link you pasted above. Could you please paste a github link?

**Reply by Y-T-G** [reply]
You can try this

https://old-reddit-com.translate.goog/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/?_x_tr_sl=es&_x_tr_tl=en&_x_tr_hl=en-GB

**Reply by ardeal** [reply]
Got it. Thank you!

**Reply by glenn-jocher** [reply]
You're welcome! If you have any further questions about YOLO or training configurations, feel free to ask.

**Reply by sayyaradonis1** [reply]
Hi dear glenn,
I would like to continue training yolov11 model that is trained by coco dataset for example yolo11s.pt  for only person detection by some custom person dataset.
How should I do?

**Reply by glenn-jocher** [reply]
To continue training `yolo11s.pt` for person detection, ensure your dataset is in YOLO format with only the "person" class labeled. Then, use the following command: `yolo detect train data=your_dataset.yaml model=yolo11s.pt epochs=50 imgsz=640`. Adjust parameters as needed. For more details, refer to the [Train documentation](https://docs.ultralytics.com/modes/train/).

**Reply by sayyaradonis1** [reply]
Thank you for your guidance. 
We used RoboFlow to label the car, and all the labels in the annotation file were set to zero, while in Coco, the car class is 3. please help.

**Reply by glenn-jocher** [reply]
@sayyaradonis1 it seems your dataset annotations are misaligned with the COCO class indices. Ensure that the class indices in your annotation files match the COCO format. You can use a script to update the indices or reconfigure them in RoboFlow. For details on annotation formats, refer to [Ultralytics Dataset Guide](https://docs.ultralytics.com/datasets/detect/).

**Comment by ardeal** [comment]
@glenn-jocher 

I use the following code to re-train yolov11. 
The questions are:
1) the weights are initialized by yolov11m.pt, right?
2) are those data augments enabled in default config?

```python
model_path = "yolo11m.pt"
model = YOLO(model_path)

device = torch.cuda.device(0)

train_results = model.train(
	data=r'cfg\datasets\person_ebike_bike_debug.yaml',
	epochs=100,  # number of training epochs
	imgsz=640,  # training image size
	device='0',  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
)

metrics = model.val()
```

**Reply by glenn-jocher** [reply]
@ardeal yes, the weights are initialized by `yolo11m.pt` when you load the model using `YOLO(model_path)`. Regarding data augmentation, standard augmentations (e.g., mosaic, mixup, color, etc.) are enabled by default in the training configuration. You can adjust or disable these augmentations by modifying the `augmentations` section in your dataset configuration or passing specific arguments during training. For more details, check the [Train Documentation](https://docs.ultralytics.com/modes/train/). Let us know if you have further questions!

**Comment by ardeal** [comment]
@glenn-jocher 
Thank you for your reply!

I did the following experiment:
### step1: choose all those COCO 2017images which contains person, or bycycle, or motobicycle. I changed label of person, motobicycle and bike to 0, 1, and 2.
```python
image  count == 65382
object count == 272962
person count == 257252
ebike  count == 8654
bike   count == 7056
```

### step2: I prepare my own dataset:
```python
image  count == 18798
object count == 38247
person count == 17126
ebike  count == 15252
bike   count == 5869
```

### step3: I put the above 2 datasets in one folder.
### step4: I modified yolov11 configuration: nc=3, warmup_epochs = 0, lr0=0.01/10
```python
nc: 3
names:
  0: person
  1: ebike
  2: bike
  
```
### step5: train yolov11 with the following code:
```python
    model_path = r"yolo11m.pt"
    model = YOLO(model_path)
    device = torch.cuda.device(0)
    train_results = model.train(
        data=r'cfg\datasets\person_ebike_bike.yaml',  # path to dataset YAML
        epochs=100,  # number of training epochs
        imgsz=640,  # training image size
        device='0',  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
    )
    metrics = model.val()
```


### results: the bad result is: the trained model usually recognize table legs as person :) 
![image](https://github.com/user-attachments/assets/5bbcb592-ee23-4dfe-b029-3f9888a5e396)
![image](https://github.com/user-attachments/assets/7b0b36ed-affe-4249-a89b-d243eeab3c10)

**Reply by glenn-jocher** [reply]
@ardeal it seems like the model is overfitting or encountering label-related confusion. Ensure your dataset is well-balanced and accurately labeled. Additionally, you may want to apply data augmentation techniques (e.g., flipping, scaling) to increase diversity and reduce overfitting. If the issue persists, consider fine-tuning learning rate, adding more training data, or increasing warmup_epochs. For further guidance, refer to the [YOLO11 training documentation](https://docs.ultralytics.com/modes/train).
